osf/models/contributor.py

```diff
@@ -41,6 +42,14 @@ class Meta:
         # NOTE: Adds an _order column
         order_with_respect_to = 'node'
 
+    def save(self, *args, **kwargs):
+        if not self.user.is_institutional_admin_or_curator():
+            return super().save(*args, **kwargs)
+        elif self.visible:
+            raise IntegrityError('Curators cannot be made bibliographic contributors')
+        else:
+            return super().save(*args, **kwargs)
```

Comment:
I think this could be simplified by saying `if self.user.is_institutional_admin_or_curator() and self.visible`, then just elsing the super call.